src/frida/repl.py

```diff
@@ -78,15 +89,37 @@ def _unload_script(self):
 
         def _process_input(self, reactor):
             self._print_startup_message()
-            self._ready.wait()
+            while self._ready.wait(0.5) != True:
+                if not reactor._running:
```

Comment:
Why? We are only reading `_running`, which can't change from `False` to `True` at this point anyway, so when we read a `False` we know we have to stop.